### DISCUSSION TITLE
Nested formatting inside of List spacers?

### DISCUSSION THREAD

**Discussion by karljj1** [axuno/SmartFormat#276] [Ideas]
Hey,

I had one of our users ask if it was possible to apply formatting to the spacer in a List.
E.G
"{Names:l:{}|, | {Not:nor|or} }"

Names is a list of string
Not is a boolean

It's giving a result like "John, Mary Amy", expecting "John, Mary nor Amy"

It does seem that the spacer is just treated as literal text, do you think its worth supporting formatting here?

**Comment by axunonb** [comment]
Hi Karl, good to hear from you!
Yes, spacers are actually parsed as `Format` (of course), and with your example, the last item has `Format.HasNested == true`.
But we only use `Format.GetLiteralText()` to process the spacers as `string`s. That's why the `Placeholder` is just discarded.
So I would say, instead of `spacerFormattingInfo.Write(spacer)` we had to implement `spacerFormattingInfo.FormatAsChild(...)`.
Wouldn't this be a nice PR? :)

**Reply by karljj1** [reply]
Ah interesting. Maybe ill make that PR ;)

**Reply by axunonb** [reply]
[Converted to issue #279](https://github.com/axuno/SmartFormat/issues/279)
